src/views/Layout.jsx

```diff
@@ -11,19 +16,113 @@ import './Layout.css';
  * defined in `App.jsx`.
  */
 
-export function Layout() {
+export function Layout({ data, listToken, setAllDetailsOpen }) {
+	const location = useLocation();
+	const currentPath = location.pathname;
+
+	const [showListToken, setShowListToken] = useState(false);
+	const [copiedToken, setCopiedToken] = useState(false);
+
+	function toggleListToken() {
+		setShowListToken((prevState) => !prevState);
+	}
+
+	function copyTokenToClipboard() {
+		window.navigator.clipboard.writeText(listToken);
+		setCopiedToken(true);
+
+		setTimeout(() => {
+			setShowListToken(false);
+			setCopiedToken(false);
+		}, 2000);
+	}
+
+	function openAllItemDetails() {
+		setAllDetailsOpen((prevState) => !prevState);
+	}
+
 	return (
 		<>
-			<div className="Layout h-screen max-w-md mx-auto overflow-clip bg-white">
-				<header className="Layout-header">
-					<h1>Smart shopping list</h1>
-				</header>
+			<div className="h-screen md:max-w-md md:mx-auto">
+				{listToken && (
+					<header className="h-auto my-8">
+						<div className="flex items-center gap-4">
+							<img
+								src="/img/logo-header.svg"
+								alt="Shroomy logo"
+								className="w-10"
+							/>
+							<h1 className="font-logo text-3xl mr-auto">Shroomy</h1>
+							{currentPath !== '/about' && (
+								<button
+									className="tooltip flex justify-center"
+									onClick={toggleListToken}
+									aria-describedby="share-button-tooltip"
+								>
+									<FontAwesomeIcon
+										icon={faArrowUpFromBracket}
+										className="h-5 hover:text-main"
+									></FontAwesomeIcon>
+									<span
+										className="tooltiptext icon mt-1"
+										id="share-button-tooltip"
+									>
+										Share with others
+									</span>
+								</button>
+							)}
+						</div>
+						{currentPath !== '/about' && (
+							<>
+								<p className="text-center mt-4">
+									You have{' '}
+									<span className="text-main font-bold">
+										{data.length} items
+									</span>{' '}
+									in your shopping list
+								</p>
+								<div className="flex justify-center gap-2 mt-1">
+									<button
+										className="text-sm font-bold hover:text-main hover:cursor-pointer"
+										onClick={openAllItemDetails}
+									>
+										Show all item details
+									</button>
+									<div>|</div>
+									<Link
+										to="/about"
+										className="text-sm font-bold hover:text-main"
```

Comment:
@nesha centering the text vertically. 

```suggestion
										className="text-sm font-bold hover:text-main my-auto"
```